impl/helpers.go

```diff
@@ -49,8 +49,8 @@ func (r *ReliableRedisStreamClient) lockAndExtend(mutex *redsync.Mutex) error {
 }
 
 func (r *ReliableRedisStreamClient) safeCloseLBS() {
-	if !r.lbsChanClosed {
+	if !r.lbsChanClosed.Load() {
 		close(r.lbsChan)
-		r.lbsChanClosed = true
+		r.lbsChanClosed.CompareAndSwap(false, true)
 	}
```

Comment:
The race still exists. Mutliple threads can execute line 52 and enter before 54 executes. It should be

```suggestion
	if r.lbsChanClosed.CompareAndSwap(false, true) {
		close(r.lbsChan)
	}
```